File: task_equivalents/llm_translations/llama/base/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
import pandas as pd
from sklearn.model_selection import train_test_split

# Load the data
College = pd.read_csv('data/csv/College.csv')
College.set_index(College.columns[0], inplace=True)

# Create a new column 'Private01' and drop 'Private'
College['Private01'] = College['Private'].map({'Yes': 1, 'No': 0})
College.drop('Private', axis=1, inplace=True)

# Split data into features (X) and target (Y)
X = College.drop('Apps', axis=1).apply(pd.to_numeric)
Y = College['Apps']

# Split data into training and test sets
xtrain, xtest, ytrain, ytest = train_test_split(X, Y, test_size=0.5, random_state=1)

File: task_equivalents/llm_translations/llama/base/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
library(readr)
library(dplyr)

College <- read_csv('data/csv/College.csv')
school_name <- College[, 1]
rownames(College) <- school_name
College <- College[, -1]

College <- College %>% 
  mutate(Private01 = ifelse(Private == 'Yes', 1, 0)) %>% 
  select(-Private)

X <- College %>% 
  select(-Apps) %>% 
  mutate(across(everything(), as.numeric))
Y <- College$Apps

set.seed(1)
train_index <- sample(nrow(X), nrow(X) / 2)
xtrain <- X[train_index, ]
xtest <- X[-train_index, ]
ytrain <- Y[train_index]
ytest <- Y[-train_index]